example/particle_leapfrog/particle_leapfrog.cpp

```diff
@@ -276,6 +276,9 @@ TaskStatus TransportParticles(MeshBlock *pmb, const StagedIntegrator *integrator
           x(n) += vx(n) * 0.5 * dt;
           y(n) += vy(n) * 0.5 * dt;
           z(n) += vz(n) * 0.5 * dt;
+
+          bool on_current_mesh_block = true;
+          swarm_d.GetNeighborBlockIndex(n, x(n), y(n), z(n), on_current_mesh_block);
```

Comment:
This call is necessary because it sets an internal `Swarm` variable saying that this particle is off the current meshblock and boundary conditions/communication need to be applied to it. This is not great design (at *least* I should have added a comment to my example saying why this needs to be there) but I did it this way to avoid having the `Swarm` internally call boundary checking functions many times.

Reply:
I see. In that case I suggest to rename the function to expose that it also make some internal changes (which I naively would not expect from a `Get` function.
Similarly, that function is currently marked `const`, which I'm now also wondering if it's semantically correct as it acutally updates the content of the private `blockIndex_`.